pkg/build/gobuild_test.go

```diff
@@ -772,6 +774,42 @@ func TestGoBuild(t *testing.T) {
 	})
 }
 
+func TestGoBuildWithKOCACHE(t *testing.T) {
+	now := time.Now() // current local time
+	sec := now.Unix()
+	koCacheDir := "/tmp/"
```

Comment:
I think `/tmp/` might not always be writable. I'd set this to https://pkg.go.dev/testing#T.TempDir and ensure that the `filepath.Join(t.TempDir(), strconv.FormatInt(sec, 10))` directory exists.

Reply:
thank you, great advice, done. 🙋🏻‍♂️